google/gax/retry.py

```diff
@@ -142,6 +142,16 @@ def inner(*args):
                     timeout = min(
                         timeout * timeout_mult, max_timeout, deadline - now)
 
-        raise exc
+                # If we have passed our deadline, raise the exception.
+                if deadline is not None and now >= deadline:
+                    exc = errors.RetryError(
+                        'Retry timeout exceeded; exception: %s' % exception,
+                        cause=exception,
+                    )
+                    six.reraise(errors.RetryError, exc, sys.exc_info()[2])
```

Comment:
@geigerj This is something I would like a second opinion on -- what I am doing here is raising the `RetryError` with the given message (which now uses the exception's message), but maintaining _the original_ traceback. I assert this is more useful for developers, and reduces the need to make people care about `.cause.exception`.